Swiftfin/Views/AdminDashboardView/ServerDevices/DeviceDetailsView/DeviceDetailsView.swift

```diff
@@ -12,101 +12,94 @@ import SwiftUI
                     user: user,
-                    lastActivityDate: viewModel.device.dateLastActivity
+                    lastActivityDate: device.dateLastActivity
                 ) {
                     router.route(to: .userDetails(user: user))
                 }
             }
 
-            CustomDeviceNameSection(customName: $temporaryCustomName)
+            Section(L10n.name) {
+                TextField(
+                    L10n.customName,
+                    text: $temporaryCustomName
+                )
+            }
 
             AdminDashboardView.DeviceSection(
-                client: viewModel.device.appName,
-                device: viewModel.device.name,
-                version: viewModel.device.appVersion
+                client: device.appName,
+                device: device.name,
+                version: device.appVersion
             )
 
-            CapabilitiesSection(device: viewModel.device)
+            CapabilitiesSection(device: device)
         }
         .navigationTitle(L10n.device)
         .onReceive(viewModel.events) { event in
             switch event {
-            case let .error(eventError):
-                UIDevice.feedback(.error)
-                error = eventError
-            case .setCustomName:
+            case .updatedCustomName:
                 UIDevice.feedback(.success)
-                isPresentingSuccess = true
+            default:
+                break
             }
         }
         .topBarTrailing {
             if viewModel.backgroundStates.contains(.updating) {
                 ProgressView()
             }
             Button(L10n.save) {
-                UIDevice.impact(.light)
-                if viewModel.device.id != nil {
-                    viewModel.send(.setCustomName(temporaryCustomName))
+                if let id = device.id {
+                    viewModel.update(
+                        id: id,
+                        options: .init(
+                            customName: temporaryCustomName
+                        )
+                    )
+
+                    device.customName = temporaryCustomName
                 }
             }
             .buttonStyle(.toolbarPill)
-            .disabled(temporaryCustomName == viewModel.device.customName)
-        }
-        .alert(
-            L10n.success.text,
-            isPresented: $isPresentingSuccess
-        ) {
-            Button(L10n.dismiss, role: .cancel)
-        } message: {
-            Text(L10n.customDeviceNameSaved(temporaryCustomName))
+            .disabled(temporaryCustomName.isEmpty || temporaryCustomName == device.customName)
         }
-        .errorMessage($error)
+        .errorMessage($viewModel.error)
+    }
+}
+
+extension DeviceDetailsView {
+
+    init(device: DeviceInfoDto, viewModel: DevicesViewModel) {
+        self.device = device
+        self.viewModel = viewModel
+        self.temporaryCustomName = device.customName ?? ""
+    }
+
+    init(device: DeviceInfoDto) {
```

Comment:
At least for now, we don't have a use case where device details wouldn't be presented without a view model - so we can remove this init. Additionally, this kind of functionality would have had to use a `StateObject` instead.